src/Interpreters/executeQuery.cpp

```diff
@@ -859,6 +903,15 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
                                     pulling_pipeline = pipeline.pulling(),
                                     query_span](QueryPipeline & query_pipeline) mutable
             {
+                /// Write query result into query result cache (if enabled)
+                auto query_result_cache = context->getQueryResultCache();
+                if (query_result_cache != nullptr
+                    && pulling_pipeline
+                    && enable_experimental_query_result_cache)
```

Comment:
shouldn't we also add `&& (!astContainsNonDeterministicFunctions(ast, context) || settings.query_result_cache_store_results_of_queries_with_nondeterministic_functions)`?

Reply:
Sure, fixed.

(`QueryPipeline::finalizeWriteInQueryResultCache` would have done nothing if there is no QC transform in the pipeline, but it's better to make things explicitly)